Improve codecov relate commands

tanjieling0 opened the pull request:
Improved codecov for the following
<img width="783" alt="Screenshot 2024-04-13 at 7 23 55 PM" src="https://github.com/AY2324S2-CS2103T-F12-1/tp/assets/122504209/790fed74-4bb0-45ec-91f6-4b3f520b7d0e">

tanjieling0 committed:
Improve codecov for relate command

tanjieling0 committed:
Improve codecov unrelate and idcontainsdigitspredicate

tanjieling0 committed:
Update relate command tests

tanjieling0 committed:
Add tests for logic manager

tanjieling0 committed:
Add relate command tests

shaunlxw commented on src/test/java/seedu/address/model/person/filter/IdContainsDigitsPredicateTest.java:
```diff
@@ -0,0 +1,60 @@
+
+public class IdContainsDigitsPredicateTest {
+
+    @Test
+    public void getIds_test_success() {
+        IdContainsDigitsPredicate predicate = new IdContainsDigitsPredicate(List.of(ID_FIRST_PERSON.value,
+                ID_SECOND_PERSON.value));
+        assertEquals(List.of(ID_FIRST_PERSON.value, ID_SECOND_PERSON.value), predicate.getIds());
+    }
```
Seems like `getIds` is only used in this test. Maybe we can entirely remove that method.

shaunlxw commented on src/test/java/seedu/address/logic/commands/RelateCommandTest.java:
```diff
@@ -71,6 +71,32 @@ public void execute_invalidSecondPersonId_throwsCommandException() {
+        IdContainsDigitsPredicate predicate = new IdContainsDigitsPredicate(List.of(ID_FIRST_PERSON.value,
+                ID_FIRST_PERSON.value));
+        RelateCommand command = new RelateCommand(predicate);
+        assertThrows(CommandException.class, () -> command.execute(model));
```
Could possible use `assertThrows(Throwable, Executable, String)` or `assertCommandFailure(...)` to also verify the error message displayed to user.

shaunlxw commented on src/test/java/seedu/address/logic/commands/RelateCommandTest.java:
```diff
@@ -71,6 +71,32 @@ public void execute_invalidSecondPersonId_throwsCommandException() {
+                ID_SECOND_PERSON.value));
+        RelateCommand command = new RelateCommand(predicate);
+
+        assertThrows(CommandException.class, () -> command.execute(model));
```
Same as above, can verify error message too.

shaunlxw reviewed (approved):
LGTM! 👍

shaunlxw committed:
IdContainsDigitsPredicate: remove #getIds

Change from assertThrows(...) to assertCommandFailure(...)
in RelateCommandTest and UnrelateCommandTest
